woocommerce/wc-admin
# Add Coupons autocompleter and relevant configs

Add Coupons to Advanced Filters following established patterns.

<img width="891" alt="screen shot 2018-09-18 at 1 17 39 pm" src="https://user-images.githubusercontent.com/1922453/45658801-4932be80-bb45-11e8-9473-1e6b23bcef1a.png">

## Additional changes
 
* Coupons config including a `getCouponLabelsById` function
* Coupons report page
* Coupons autocompleter
* Removed unused stylesheets on orders and products reports

## Test
1. `/wp-admin/admin.php?page=wc-admin#/analytics/coupons?filter=compare`
2. See basic filters appear on the new Coupons report, including a Compare card
3. Make sure you have a few coupons on your test site and search for them
4. Ensure the basic compare functionality is present, including URL updates and tags appearing on page refresh
5. Do the same with Advanced Filters at `/wp-admin/admin.php?page=wc-admin#/analytics/orders?filter=advanced` by adding Coupon Codes filter.

Fixes https://github.com/woocommerce/wc-admin/issues/342
